Signal/src/ViewControllers/ColorPickerViewController.swift

```diff
@@ -0,0 +1,143 @@
+//
+//  Copyright (c) 2018 Open Whisper Systems. All rights reserved.
+//
+
+import Foundation
+
+let colorSwatchHeight: CGFloat = 60
+
+class ColorView: UIView {
+    let color: UIColor
+    let swatchView: UIView
+
+    required init(color: UIColor) {
+        self.color = color
+        self.swatchView = UIView()
+
+        super.init(frame: .zero)
+
+        swatchView.backgroundColor = color
+
+        self.swatchView.layer.cornerRadius = colorSwatchHeight / 2
+
+        self.addSubview(swatchView)
+
+        swatchView.autoVCenterInSuperview()
+        swatchView.autoSetDimension(.height, toSize: colorSwatchHeight)
+        swatchView.autoPinLeadingToSuperviewMargin()
+        swatchView.autoPinTrailingToSuperviewMargin()
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+}
+
+@objc
+protocol ColorPickerDelegate: class {
+    func colorPickerDidCancel(_ colorPicker: ColorPickerViewController)
+    func colorPicker(_ colorPicker: ColorPickerViewController, didPickColorName colorName: String)
+}
+
+@objc
+class ColorPickerViewController: UIViewController, UIPickerViewDelegate, UIPickerViewDataSource {
+
+    private let pickerView: UIPickerView
+    private let thread: TSThread
+    private let colors: [UIColor]
+
+    @objc public weak var delegate: ColorPickerDelegate?
+
+    @objc
+    required init(thread: TSThread) {
+        self.thread = thread
+        self.pickerView = UIPickerView()
+        self.colors = UIColor.ows_conversationColors
+
+        super.init(nibName: nil, bundle: nil)
+
+        self.navigationItem.leftBarButtonItem = UIBarButtonItem(barButtonSystemItem: .cancel, target: self, action: #selector(didTapCancel))
+        self.navigationItem.rightBarButtonItem = UIBarButtonItem(barButtonSystemItem: .save, target: self, action: #selector(didTapSave))
+
+        pickerView.dataSource = self
+        pickerView.delegate = self
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func loadView() {
+        self.view = UIView()
+        view.backgroundColor = .white
+        view.addSubview(pickerView)
+
+        pickerView.autoVCenterInSuperview()
+        pickerView.autoPinLeadingToSuperviewMargin()
+        pickerView.autoPinTrailingToSuperviewMargin()
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        if let colorName = thread.conversationColorName,
+            let currentColor = UIColor.ows_conversationColor(colorName: colorName),
+            let index = colors.index(of: currentColor) {
```

Comment:
One has to wonder how UIColor comparison is done and whether this is reliable.  Would it be safer to replace `colors` with `colorNames` and compare color names here?